src/core/Akka.Persistence.Tests/PersistentActorRecoveryTimeoutSpec.cs

```diff
@@ -11,14 +11,16 @@
 using Akka.Event;
 using Akka.Persistence.Tests.Journal;
 using Akka.TestKit;
+using Akka.Util.Internal;
 using Xunit;
 
 namespace Akka.Persistence.Tests
 {
-
+    [Collection("PersistentActorRecoveryTimeout")] // force tests to run sequentially
     public class PersistentActorRecoveryTimeoutSpec : PersistenceSpec
     {
-        private const string JournalId = "persistent-actor-recovery-timeout-spec";
+        private static readonly AtomicCounter JournalIdNumber = new AtomicCounter(0);
```

Comment:
Race conditions in this test were caused by multiple parallel test cases accessing the same `SteppingMemoryJournal` - forced this test case to run sequentially (via the `Collection` attribute on the class) and made it so each `SteppingMemoryJournal` instance was isolated to each test case.